incubator/sparkoperator/templates/spark-operator-rbac.yaml

```diff
@@ -0,0 +1,52 @@
+{{ if not .Values.enableWebhook }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .Values.operatorNamespace }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: sparkoperator
+  namespace: {{ .Values.operatorNamespace }}
+{{ end }}
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: sparkoperator
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["*"]
+- apiGroups: [""]
+  resources: ["services", "configmaps"]
+  verbs: ["create", "get", "delete"]
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["get"]
+- apiGroups: [""]
+  resources: ["events"]
+  verbs: ["create", "update", "patch"]
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["create", "get", "update", "delete"]
+- apiGroups: ["admissionregistration.k8s.io"]
+  resources: ["mutatingwebhookconfigurations"]
+  verbs: ["create", "get", "update", "delete"]
+- apiGroups: ["sparkoperator.k8s.io"]
+  resources: ["sparkapplications", "scheduledsparkapplications"]
+  verbs: ["*"]
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: sparkoperator
+subjects:
+  - kind: ServiceAccount
+    name: sparkoperator
+    namespace: {{ .Values.operatorNamespace }}
```

Comment:
how is this handled if webhooks are used? Meaning the order?

Reply:
The line that you referred to here would only run when the webhook is not enabled (see the first line in the file). When it is enabled, the order is enforced using [hook weights](https://github.com/helm/charts/pull/6239/files#diff-b72874954241f266b2c52983308e8dbcR11). Note the -10, -5 and 0.